syntax="proto3";
package telemetry;
import "enums.proto";
import "uuid.proto";

message Telemetry {
  enums.ModuleStatusType emg_status = 1;
  enums.ModuleStatusType display_status = 2;
  enums.ModuleStatusType gyro_status = 3;
  enums.DriverStatusType driver_status = 4;
  int64 last_time_sync = 5;
  int32 emg = 6;
  uuid.UUID executable_gesture = 7;
  int32 power = 8;
  int32 pointer_finger_position = 9;
  int32 middle_finger_position = 10;
  int32 ring_finger_position = 11;
  int32 little_finger_position = 12;
  int32 thumb_finger_position = 13;
}

message GetTelemetry {
  Telemetry telemetry = 1;
}

message StartTelemetry {
  int32 interval_ms = 1;
}
